CWE/inventions/New inventions.txt

```diff
@@ -300,4 +300,465 @@ full_body_replacement = {
+			factor = 1
+			strd_internet = 1
+		}
+		modifier = {
+			factor = 1
+			beijing_consensus = 1
+		}
+		modifier = {
+			factor = 1
+			OR = { ruling_party_ideology = socialist ruling_party_ideology = progressive ruling_party_ideology = liberal }
+		}
+		modifier = {
+			factor = 1
+			OR = { economic_policy = planned_economy1 economic_policy = planned_economy }
+		}
+	}
+	effect = {
+		factory_output = 0.01
+	}
+}
+
+web_1 = {
+	limit = { strd_internet = 1 } #Internet Service Providers
+	chance = {
+		base = 1
+		modifier = {
+			factor = 1
+			culture_immigration = 1
+		}
+	}
+	effect = {
+		shared_prestige = 10
+		factory_goods_output = {
+			culture = 0.01
+		}
+	}
+}
+
+
+web_2 = {
+	limit = { high_speed_uni_network = 1 } #High Speed Internet
+	chance = {
+		base = 1
+		modifier = {
+			factor = 1
+			digital_revolution = 1
+		}
+	}
+	effect = {
+		shared_prestige = 10
+		factory_goods_output = {
+			culture = 0.01
+		}
+	}
+}
+
+
+web_3 = {
+	limit = { uni_network = 1 } #Ultra-High Speed Internet
+	chance = {
+		base = 1
+		modifier = {
+			factor = 1
+			social_networking = 1
+		}
+		modifier = {
+			factor = 1
+			beijing_consensus = 1
+		}
+	}
+	effect = {
+		shared_prestige = 15
+		factory_goods_output = {
+			culture = 0.01
+		}
+	}
+}
+
+
+new_left = {
+	limit = { wars_of_national_liberation = 1 } #Ideological Mass Mobilization
+	chance = {
+		base = 1
+		modifier = {
+			factor = 1
+			OR = { government = democracy government = democracy1 government = hms_government government = hms_government1 government = socialist_democracy }
+		}
+		modifier = {
+			factor = 1
+			OR = { continent = europe continent = north_america }
+		}
+		modifier = {
+			factor = -1
+			OR = { ruling_party_ideology = communist ruling_party_ideology = communist_social }
+		}
+		modifier = {
+			factor = -2
+			NOT = { year = 1960 }
+		}
+		modifier = {
+			factor = 1
+			has_country_flag = eurocommunism
+		}
+	}
+	effect = {
+		social_reform_desire = 0.01
+		rebel_org_gain = {
+			faction = communist_rebels
+			value = 0.05
+		}
+	}
+}
+
+eurocommunism_inv = { 
+	limit = { seventies_subcultures  = 1 } #Progressivism
+	chance = {
+		base = 1
+		modifier = {
+			factor = 1
+			OR = { government = democracy government = democracy1 government = hms_government government = hms_government1 government = socialist_democracy }
+		}
+		modifier = {
+			factor = -1
+			NOT = { continent = europe }
+		}
+		modifier = {
+			factor = -1
+			OR = { ruling_party_ideology = communist ruling_party_ideology = communist_social }
+		}
+		modifier = {
+			factor = 3
+			has_country_flag = eurocommunism
+		}
+	}
+	effect = {
+		social_reform_desire = 0.01
+	}
+}
+
+human_immunocontraception = { 
+	limit = { genomics = 1 } #AI Guided Manufacturing
+	chance = {
+		base = 1
+		modifier = {
+			factor = -1
+			OR = { ruling_party_ideology = traditionalist ruling_party_ideology = populist ruling_party_ideology = nationalist }
+		}
+	}
+	effect = {
+        	pop_growth = -0.0001
```

Comment:
Sorry, I don't allow techs to affect population growth as that means I have to rebalance it again.